ferrocene/doc/internal-procedures/src/manual-upstream-pulls.rst

```diff
@@ -11,96 +11,60 @@ An "upstream pull" is a pull request that pulls in commits from the upstream Rus
-`<UPSTREAM_BRANCH>` branch, create a commit to remove the excluded files
-(see below) from the upstream branch, and finally try to merge upstream into
-the current branch.
-
-The script is able to resolve some kinds of merge conflicts on its own. If
-there are still remaining merge conflicts after the automatic resolution the
-script will exit. You'll have to fix the merge conflicts yourself in that case:
-
-There are two kinds of conflicts you will encounter in the output that the
-script produces:.
-
-* Submodule conflict ::
-
-    Please manually handle the merging of each conflicted submodule.
-    This can be accomplished with the following steps:
-    - go to submodule (src/tools/cargo), and either merge commit <COMMIT_HASH>
-      or update to an existing commit which has merged those changes
-    - come back to superproject and run:
-
-          git add src/tools/cargo
-
-      to record the above merge or update
-    - resolve any other conflicts in the superproject
-    - commit the resulting index in the superproject
-
-  The message already describes how to fix this conflict. The commands required
-  for this example are::
+* Submodule conflict
+  To fix a submodule conflict run::
 
     cd src/tools/cargo
     git merge <COMMIT_HASH>
     cd ../../..
     git add src/tools/cargo
 
-* General conflicts::
+  <COMMIT_HASH> is ...
+
+* File deleted by one side of the merge
+  This happens when we made changes to a file and it gets deleted by upstream. To
+  resolve this, remove the file::
 
-    CONFLICT (content): Merge conflict in src/bootstrap/flags.rs
+    git rm <FILE_1> <FILE_2> # removes file(s) and stages that change
 
+  If this file was a test case used in the traceability matrix, we might need to
+  update the traceability matrix, if some section is missing a test case. But finding
+  a new testcase should be done in a separate PR.
+
+* General conflict
   These are more common conflicts and will require manual fixing depending on
   the conflict in source.
 
-After having fixed the conflicts, finish up the merge by doing the following::
+After having fixed the conflicts, add a new commit and push it to the PR::
 
   git add . # add your changes to the staging area
-  git merge --continue # finish the merge
-  git push --set-upstream origin <NEW_PULL_BRANCH> # push out the pull request branch
-
-Finally generate the PR description with::
-
-  ferrocene/tools/pull-upstream/generate_pr_body.py origin <BASE_BRANCH> <NEW_PULL_BRANCH>
+  git commit
+  git push
 
-This will output a markdown list of the pull requests that have been pulled
-from the upstream Rust repository. Paste that into the PR description of
-the pull request you create from your `<NEW_PULL_BRANCH>` branch.
-You may omit `<NEW_PULL_BRANCH>` in which case the script will use the currently checked
-out branch.
+Now another developer needs to sign off the PR (see "Sign off Upstream Pull" section above).
```

Comment:
The section above doesn't mention this :slightly_smiling_face: (even though this sentence is correct).

Reply:
Do you mean the "Sign off Upstram Pull" section? Iiuc in that case there is only one developer involved. The automation opens the PR, one developer reviews it and merges it. But if there are merge conflicts, the first developer fixes them and a second developer reviews it.

Reply:
Indeed. I'm wondering why the "(see "Sign off Upstream Pull" section above)" is there then?

Reply:
Good point. It just includes the 'This will trigger the "full" CI workflow. If the workflow succeeds, the Upstream Pull is getting merged.', but I guess that might be clear.